### PR TITLE
[FEATURE REQUEST #488]Export C-style APIs for RocketMQ client

### PR DESCRIPTION
**Pull request by ShannonDing**
Please do not create a Pull Request without creating an issue first. 

## What is the purpose of the change

In order to connect RocketMQ, a c-style client is needed.
Add C style APIs.

## Brief changelog

At present, this C style APIs supports
sending message in synchronous mode
consuming message using push model

## Verifying this change

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [ ] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

**Comment by vongosling** [comment]
It would be helpful if you could finish the checklist. I would like to help you to merge the pr.

File: rocketmq-cpp/CMakeLists.txt [hunk 1]
```diff
@@ -40,8 +40,7 @@ set(CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS ON)
 set(CMAKE_VERBOSE_MAKEFILE 1)
 
 #Find dependency 
-set(BOOST_INCLUDEDIR
-/home/qiwei.wqw/sambarshare/sambashare/metaq/master/rocketmq-client4cpp/rpm/.dep_create/include/)
+set(BOOST_INCLUDEDIR /usr/local/include/)
```
Comment (vongosling) [inline rocketmq-cpp/CMakeLists.txt]: Nice work

File: rocketmq-cpp/CMakeLists.txt [hunk 2]
```diff
@@ -60,8 +59,9 @@ if(Boost_FOUND)
     include_directories(${Boost_INCLUDE_DIRS})
 endif()
 
-#set(LIBEVENT_INCLUDE_DIR C:/libevent-2.0.22-stable/include)
+#set(LIBEVENT_INCLUDE_DIR /usr/local/include/)
 set(Libevent_USE_STATIC_LIBS    ON) # only find static libs
+#set(Libevent_DIR    /usr/local/lib/) # only find static libs
```
Comment (vongosling) [inline rocketmq-cpp/CMakeLists.txt]: nice

File: rocketmq-cpp/CMakeLists.txt [hunk 3]
```diff
@@ -71,7 +71,7 @@ if(LIBEVENT_FOUND)
 endif()
 
 #set(JSONCPP_INCLUDE_DIRS C:/jsoncpp-0.10.6/include)
-set(JSONCPP_USE_STATIC_LIBS    ON) # only find static libs
+set(JSONCPP_USE_STATIC_LIBS    OFF) # only find static libs
```
Comment (vongosling) [inline rocketmq-cpp/CMakeLists.txt]: only find static libs?
I noticed you turn it off. Maybe new comments here.

Comment (ShannonDing) [reply rocketmq-cpp/CMakeLists.txt]: Recommend to use dynamic library.

**Pull request merged**